stdlib/2/typing.pyi

```diff
@@ -352,64 +352,77 @@ class TextIO(IO[unicode]):
 class ByteString(Sequence[int], metaclass=ABCMeta): ...
 
 class Match(Generic[AnyStr]):
-    pos = 0
-    endpos = 0
-    lastindex = 0
-    lastgroup = ...  # type: AnyStr
-    string = ...  # type: AnyStr
+    pos: int
+    endpos: int
+    lastindex: Optional[int]
+    string: AnyStr
 
     # The regular expression object whose match() or search() method produced
-    # this match instance.
-    re = ...  # type: Pattern[AnyStr]
+    # this match instance. This should not be Pattern[AnyStr] because the type
+    # of the pattern is independent of the type of the matched string in
+    # Python 2. Strictly speaking Match should be generic over AnyStr twice:
+    # once for the type of the pattern and once for the type of the matched
+    # string.
+    re: Pattern[Any]
+    # Can be None if there are no groups; otherwise matches the type of the pattern.
+    lastgroup: Any
```

Comment:
This is probably just a personal preference, but since `Any` is just a placeholder for a more specific type that can't be expressed at the moment, I would write this as `Optional[Any]`, even if it is functionally equivalent.